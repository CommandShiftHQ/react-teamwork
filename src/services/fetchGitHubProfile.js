const URL = 'https://mcr-codes-cohorts.herokuapp.com/users/';

const fetchGitHubProfile = (username) => {
  return fetch(`${URL}${username}`)
    .then(res => res.json());
};

export default fetchGitHubProfile;
