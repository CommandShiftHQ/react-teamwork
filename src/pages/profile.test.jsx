import React from 'react';
import { shallow } from 'enzyme';
import Profile from './profile';
import fetchGitHubProfile from '../services/fetchGitHubProfile';

jest.mock('../services/fetchGitHubProfile');
const MOCK_API_DATA = {
  repos: [{ name: 'jQuery' }],
  profile: {
    name: 'ersel',
    followers: 20,
  },
  contributions: {
    count: 2000,
  },
  events: {
    count: 1000,
  },
};
const MOCK_GITHUB_API = Promise.resolve(MOCK_API_DATA);
fetchGitHubProfile.mockImplementation(() => MOCK_GITHUB_API);

describe('Profile page', () => {
  it('should render', (done) => {
    const dummyRoute = {
      params: {
        username: 'ersel',
      },
    };
    const myProfile = shallow(<Profile match={dummyRoute} />);
    // wait for api to resolve then call jest done()
    MOCK_GITHUB_API.then(() => {
      expect(myProfile).toMatchSnapshot();
      done();
    });
  });
});
