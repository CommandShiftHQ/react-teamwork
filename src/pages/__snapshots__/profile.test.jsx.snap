// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page should render 1`] = `
<div>
  <input
    onChange={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Get Stats
  </button>
  <br />
  <br />
  <ProfileCard
    profile={
      Object {
        "followers": 20,
        "name": "ersel",
      }
    }
    username="ersel"
  />
  <RepoList
    repos={
      Array [
        Object {
          "name": "jQuery",
        },
      ]
    }
    username="ersel"
  />
  <ContributionSummary
    contributions={
      Object {
        "count": 2000,
      }
    }
    events={
      Object {
        "count": 1000,
      }
    }
    username="ersel"
  />
</div>
`;
